google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineDeploymentRunConfigurationEditor.java

```diff
@@ -114,14 +118,20 @@ public AppEngineDeploymentRunConfigurationEditor(
     updateJarWarSelector();
     userSpecifiedArtifactFileSelector.setVisible(true);
 
-    appEngineCostWarningLabel.setText(
-        GctBundle.message("appengine.flex.deployment.cost.warning",
-            COST_WARNING_OPEN_TAG,
-            COST_WARNING_HREF_OPEN_TAG,
-            COST_WARNING_HREF_CLOSE_TAG,
-            COST_WARNING_CLOSE_TAG));
-    appEngineCostWarningLabel.addHyperlinkListener(new BrowserOpeningHyperLinkListener());
-    appEngineCostWarningLabel.setBackground(editorPanel.getBackground());
+    AppEngineEnvironment environment = ((AppEngineDeployable) deploymentSource).getEnvironment();
```

Comment:
Yeah but is there a case where an AppEngineDeployable will not also be a
DeploymentSource?  I'm just wondering if you can avoid this cast by having
the constructor take AppEngineDeployable as its source.

On Wed, Jun 8, 2016 at 3:25 PM, Etan Shaul <notifications@github.com> wrote:

> In
> google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineDeploymentRunConfigurationEditor.java
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/700#discussion_r66322378>
> :
>
> > @@ -114,14 +118,20 @@ public AppEngineDeploymentRunConfigurationEditor(
> >      updateJarWarSelector();
> >      userSpecifiedArtifactFileSelector.setVisible(true);
> >
> > -    appEngineCostWarningLabel.setText(
> > -        GctBundle.message("appengine.flex.deployment.cost.warning",
> > -            COST_WARNING_OPEN_TAG,
> > -            COST_WARNING_HREF_OPEN_TAG,
> > -            COST_WARNING_HREF_CLOSE_TAG,
> > -            COST_WARNING_CLOSE_TAG));
> > -    appEngineCostWarningLabel.addHyperlinkListener(new BrowserOpeningHyperLinkListener());
> > -    appEngineCostWarningLabel.setBackground(editorPanel.getBackground());
> > +    AppEngineEnvironment environment = ((AppEngineDeployable) deploymentSource).getEnvironment();
>
> The idea is that app engine deployable sources of type DeploymentSource
> implement AppEngineDeployable. So its actually the other way around.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/700/files/a3a2318d273b9729e13906620a2014cc69c7e449#r66322378>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AHf5HdTNZ1kHl37wy0LiI2hI3jPqfROBks5qJxcngaJpZM4Im4aR>
> .
>